src/core/sys/posix/signal.d

```diff
@@ -1224,8 +1224,8 @@ else version( OpenBSD )
 }
 else version( DragonFlyBSD )
 {
-    enum SIG_CATCH = cast(sigfn_t2) 2;
-    enum SIG_HOLD = cast(sigfn_t2) 3;
+    enum SIG_CATCH = 2;
+    enum SIG_HOLD = 3;
```

Comment:
I think Sebastian meant to change this to `enum sigfn_t2 SIG_CATCH = 2;`, i.e. move the type from the cast to the enum declaration, but not sure if it makes any difference.

Reply:
@CyberShadow Thanks for reviewing the code :-)
I just made that change (a couple of minutes ago).

Reply:
>  I just made that change (10 minutes ago).

Unless you're talking about something else which you haven't pushed yet, then my comment was about the code you did just push. There is no type declaration in it. I don't know if it matters or not, I just wanted to clarify on Sebastian's behalf.

To clarify even further, your code goes:

```d
enum SIG_CATCH = 2;
```

I believe the suggested change was:

```d
enum sigfn_t2 SIG_CATCH = 2;
```

Hopefully that should clear things up.

Reply:
@CyberShadow That gives 
"Error: cannot implicitly convert expression 2 of type int to extern (C) void function(int) nothrow @nogc"

I think i need to actually revert to the original definition using the cast.

Reply:
Reverted the previous patch..

Reply:
Ping @wilzbach , @joakim-noah and @CyberShadow 
Reverted the previous patch... I think the requested change might require a little more deliberation/discussion.

Source fragment (to give some context):
```
nothrow @nogc
{
    private alias void function(int) sigfn_t2;
    private alias void function(int, siginfo_t*, void*) sigactfn_t2;
}
...
else version( DragonFlyBSD )
{
    enum SIG_CATCH = cast(sigfn_t2) 2;
    enum SIG_HOLD = cast(sigfn_t2) 3;
...
```

Reply:
The way you had it initially was fine.  We weren't sure if you could just apply the type to the left, but as you found, it needs to be cast.

Reply:
Thanks for the update 😉. Glad I rolled it back just before ibuclaw merged it 😙

Reply:
@CyberShadow As mentioned in the dlang/druntime PR, that does not work (compiler error), i reverted it to the original version using the cast. The PR has already been merged now. If it does need to change (now or in the future), it would have to be done in a seperate PR and should also involve the other OS implementations.

Reply:
Yes, I saw the earlier messages. No objections to that, of course.